modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,336 @@
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ * @return mixed $attr Attributes for the image markup.
+ */
+function dominant_color_update_attachment_image_attributes( $attr, $attachment ) {
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
+	if ( isset( $image_meta['has_transparency'] ) ) {
+		$attr['data-has-transparency'] = $image_meta['has_transparency'] ? 'true' : 'false';
+
+		$class = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
+		if ( empty( $attr['class'] ) ) {
+			$attr['class'] = '';
+		}
+		$attr['class'] .= $class;
+	}
+
+	if ( ! empty( $image_meta['dominant_color'] ) ) {
+		$attr['data-dominant-color'] = $image_meta['dominant_color'];
+		if ( empty( $attr['style'] ) ) {
+			$attr['style'] = '';
+		}
+		$attr['style'] .= '--dominant-color: #' . $image_meta['dominant_color'] . ';';
+	}
+
+	return $attr;
+}
+add_filter( 'wp_get_attachment_image_attributes', 'dominant_color_update_attachment_image_attributes', 10, 2 );
+
+/**
+ * Filter image tags in content to add the dominant color to the image.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $filtered_image The filtered image.
+ * @param string $context The context of the image.
+ * @param int    $attachment_id The attachment ID.
+ *
+ * @return string image tag
+ */
+function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $attachment_id ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment_id );
+	if ( ! is_array( $image_meta ) ) {
+		return $filtered_image;
+	}
+
+	/**
+	 * Filters whether dominant color is added to the image.
+	 * set to false inorder disable adding the dominant color to the image.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param bool $add_dominant_color_to_image Whether to add the dominant color to the image. default true.
+	 * @param int $attachment_id The image attachment ID.
+	 * @param array $image_meta The image meta data all ready set.
+	 * @param string $filtered_image The filtered image. html including img tag
+	 * @param string $context The context of the image.
+	 */
+	$check = apply_filters( 'dominant_color_img_tag_add_dominant_color', true, $attachment_id, $image_meta, $filtered_image, $context );
+	if ( ! $check ) {
+		return $filtered_image;
+	}
+
+	$data        = '';
+	$style       = '';
+	$extra_class = '';
+
+	if ( ! empty( $image_meta['dominant_color'] ) ) {
+		$data .= sprintf( ' data-dominant-color="%s"', $image_meta['dominant_color'] );
+		$style = ' style="--dominant-color: #' . $image_meta['dominant_color'] . ';" ';
+	}
+
+	if ( isset( $image_meta['has_transparency'] ) ) {
+		$transparency = $image_meta['has_transparency'] ? 'true' : 'false';
+		$data        .= sprintf( ' data-has-transparency="%s"', $transparency );
```

Comment:
See https://github.com/WordPress/performance/pull/282/files#r883083376. This needs to be
```suggestion
		$data        .= sprintf( 'data-has-transparency="%s" ', $transparency );
```